optuna/storages/_rdb/storage.py

```diff
@@ -192,15 +193,24 @@ def set_study_direction(self, study_id: int, direction: StudyDirection) -> None:
         session = self.scoped_session()
 
         study = models.StudyModel.find_or_raise_by_id(study_id, session)
-
-        if study.direction != StudyDirection.NOT_SET and study.direction != direction:
+        current_direction = tuple(
+            [d.direction for d in models.StudyDirectionModel.where_study(study, session)]
+        )[0]
+        if current_direction != StudyDirection.NOT_SET and current_direction != direction:
             raise ValueError(
                 "Cannot overwrite study direction from {} to {}.".format(
-                    study.direction, direction
+                    current_direction, direction
                 )
             )
 
-        study.direction = direction
+        session.query(models.StudyDirectionModel).filter(
+            models.StudyDirectionModel.study_id == study_id
+        ).delete()
+
+        direction_model = models.StudyDirectionModel(
+            study_id=study_id, objective_id=0, direction=direction
+        )
+        session.add(direction_model)
```

Comment:
Why do you remove the study direction entry and re-create the new one?
Can we overwrite the value of `StudyDirectionModel.direction` instead?